colossalai/tensor/colo_parameter.py

```diff
@@ -1,28 +1,35 @@
 from .colo_tensor import ColoTensor
 from .const import TensorType
 import torch
+from colossalai.tensor import TensorSpec, dist_spec
+from copy import copy
 
 
 class ColoParameter(ColoTensor):
     r"""A kind of ColoTensor to be considered as a module parameter.
 
     """
 
-    def __init__(self, *args, **kargs):
-        super().__init__(*args, **kargs)
-        self._type = TensorType.MODEL
+    def __new__(cls,
+                data: torch.Tensor,
+                requires_grad: bool = True,
+                spec: TensorSpec = TensorSpec(dist_spec.replicate())) -> 'ColoParameter':
+        if data is None:
+            data = torch.empty(0)
+        return torch.Tensor._make_subclass(cls, data, requires_grad)
 
-    def __new__(cls, *args, **kwargs):
-        t = super(ColoParameter, cls).__new__(cls)
-        t._type = TensorType.MODEL
-        return t
+    def __init__(self,
+                 data: torch.Tensor,
+                 requires_grad: bool = True,
+                 spec: TensorSpec = TensorSpec(dist_spec.replicate())) -> None:
+        self._spec = copy(spec)
+        self._type = TensorType.MODEL
+        self._graph_node = None
 
     @staticmethod
-    def init_from_torch_tensor(tensor: torch.Tensor, save_payload=True) -> 'ColoParameter':
-        colo_p = ColoParameter(*tensor.size(),
-                               dtype=tensor.dtype,
-                               requires_grad=tensor.requires_grad,
-                               pin_memory=tensor.is_pinned(),
-                               device=tensor.device,
-                               torch_tensor=tensor if save_payload else torch.empty(0))
-        return colo_p
+    def from_torch_tensor(tensor: torch.Tensor,
+                          requires_grad: bool = True,
+                          spec: TensorSpec = TensorSpec(dist_spec.replicate())) -> 'ColoParameter':
+        tensor = tensor.as_subclass(ColoParameter)
+        tensor.__init__(tensor, requires_grad=requires_grad, spec=spec)
+        return tensor
```

Comment:
Like the last question, why we still need from_torch_tensor?

Reply:
`ColoTensor.from_torch_tensor` and `ColoParameter.from_torch_tensor` take different arguments. Although we can make `from_torch_tensor` class method which takes `*args` and `**kwargs`, in order to make the function clear and friendly for IDE auto completion, I override it.